### ISSUE TITLE
BUG: some variant length dataset cannot be trained successfully

### ISSUE THREAD

**Issue by jianzongwu** [White-Link/UnsupervisedScalableRepresentationLearningTimeSeries#19]
Hi
I'm a student who is doing research about Time Series Representation Learning. I thought your paper and its code is a really a good start. Thank you for your wonderful work!

But when I run `transfer_ucr.py`, when the code runs to DodgerLoopDay dataset, it corrupts, and says the input contains 'NaN'.

I opened DodgerLoopDay dataset and see some NaNs in the TRAIN and TEST csv file, but I think that the code has already processed with variant length dataset and I don't know why it would come to an ERROR.

Then I see into the code file `scikit_wrappers.py`, and found in function encoder() (around line 370), some codes handling variant length datasets. But it seems that the code **only** handles NaNs appearing at the **end** of the data, but in DodgerLoopDay, NaNs are appearing actually in the **middle part** of the data, so I think that's why the code comes to an ERROR.

And I found that in your paper, some datasets in UCRArchive, including DodgerLoopDay, is not listed in the result chart, does it have something to do with the bug?

Sorry, my English is not so good, I hope you can understand me, I'm trying to explain it as explicitly as I can do.

**Comment by White-Link** [comment]
Hi, thank you very much for your interest in our research!

Indeed, we found that exactly three datasets contained missing values represented as NaNs: DodgerLoopDay, DodgerLoopGame and DodgerLoopWeekend. This is only indicated in the appendix of the paper and should have perhaps been more explicit, but, as our method is not directly applicable to time series containing missing data, we chose to ignore these datasets. Therefore, the error you encounter is expected!

We do not plan on supporting time series containing missing values, but we are interested in possible extensions to this end. As a workaround, you might for instance replace the missing values by zeros or removing them. This might be sufficient to learn satisfying representations for these datasets.
